traffic_router/connector/src/main/java/org/apache/traffic_control/traffic_router/protocol/RouterNioEndpoint.java

```diff
@@ -151,8 +151,9 @@ public RouterSocketProcessor(final SocketWrapperBase<NioChannel> socketWrapper,
 		*/
 		@Override
 		protected void doRun(){
-			final SocketWrapperBase<NioChannel> localWrapper = socketWrapper;
-			final NioChannel socket = localWrapper.getSocket();
+			@SuppressWarnings("PMD.CloseResource")
+			// The socket must be stored before calling super.doRun() because it sets socketWrapper to null
+			final NioChannel socket = socketWrapper.getSocket();
```

Comment:
I wonder if the issue mentioned in the comment above has been corrected yet. I think that was added to fix a version at least a couple major versions ago, so there's definitely been some time since then. The issue was basically breaking https requests, so it should be fairly straightforward to verify. I'll see if I can reproduce after removing this `RouterSocketProcessor` class.

Reply:
Yeah, I believe this entire `RouterSocketProcessor` class and overridden `createSocketProcessor` method are no longer necessary. In shadow testing, TR seemed fine either way and even seemed like it performed slightly better in terms of requests/sec without this.

Reply:
Neat, removed in 0d2cc14ee5.